src/main.sh

```diff
@@ -64,6 +64,11 @@ EOF
 }
 
 function installTerraform {
+  if [[ "${tfVersion}" == "latest" ]]; then
+    echo "Checking the latest version of Terraform"
+    tfVersion=$(curl -s https://releases.hashicorp.com/terraform/ | grep -o "terraform_[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*[^-]" | grep -m 1 -o "[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*")
```

Comment:
Instead of parsing the HTML content with regular expressions, there's a JSON endpoint that already has the necessary data. Parsing the JSON data like so might be more efficient.

```suggestion
    tfVersion=$(curl -sL https://releases.hashicorp.com/terraform/index.json | jq -r .versions[].version | grep -v '[-].*' | sort -rV | head -n 1)
```

The only thing I'm thinking about is whether or not to perform a check to see if `tfVersion` is empty before proceeding with the rest of this function.

Reply:
@sudomateo Thank you, I didn't know the JSON endpoint exists! 
Just pushed changes to 1) use the JSON endpoint instead of parsing HTML, and 2) check `tfVersion` after resolving the latest version then abort if empty.